### ISSUE TITLE
Offer endpoints performance

### ISSUE THREAD

**Issue by adimango** [apaleo/api#394]
## Question

Is there any response time range for the offer response?
Is it normal that we get a response back on average after 2s?

## Background

How can we get faster response for offers?

Below some logs:
```
[26/8/2020 @ 13:51:44:240] https://api.apaleo.com/booking/v1/offers?propertyId=TAN5&arrival=2021-05-30&departure=2021-05-31&adults=1 - 2165 ms
[26/8/2020 @ 13:51:44:256] https://api.apaleo.com/booking/v1/offers?propertyId=ANT5&arrival=2021-05-30&departure=2021-05-31&adults=1 - 2182 ms
[26/8/2020 @ 13:51:44:278] https://api.apaleo.com/booking/v1/offers?propertyId=FRU20&arrival=2021-05-30&departure=2021-05-31&adults=1 - 2204 ms
[26/8/2020 @ 13:51:44:763] https://api.apaleo.com/booking/v1/offers?propertyId=HOY26&arrival=2021-05-30&departure=2021-05-31&adults=1 - 2688 ms
[26/8/2020 @ 13:51:47:762] https://api.apaleo.com/booking/v1/offers?propertyId=JOR16&arrival=2021-05-30&departure=2021-05-31&adults=1 - 5687 ms
```

**Comment by dario-hd** [comment]
Hi,

I checked your setup and I saw that you have more than 100 rate plans configured in those properties (170 in one specific property). Since the offers are calculated in real-time, such latency is kinda expected. It also seems that you are trying to build something on account level since those requests are being fired within the same second.

There's another way to build such a thing. You can read the following guide even though is written for a specific use case: https://apaleo.dev/guides/business-cases/channel-integration.

Basically, the idea is to use the distribution API to subscribe to ARI. In this way, you can receive updates on rates and availability and store this data on your side for faster access.

I hope this helps.

**Comment by nziehn** [comment]
Hi @dario-hd ,

thanks for getting back to us - almost all of them have a voucher code associated to them. Can we somehow avoid going through those on your end, since this would be a drastic reduction in count?

**Comment by dario-hd** [comment]
Those shouldn't be considered already. Are requesting those offers from a specific channel e.g. IBE? To reduce the rate plans to consider you could filter by a specific channel.

**Comment by nziehn** [comment]
Hi @dario-hd,

the request that we're sending, e.g.:
https://api.apaleo.com/booking/v1/offers?propertyId=TAN5&arrival=2021-05-30&departure=2021-05-31&channelCode=Ibe&adults=1

also takes 2.3 sec but there are only 28 rate plans that fulfil:

a) correct channelCode
b) no promoCode
c) isBookable

**Comment by dario-hd** [comment]
Hi, as I said the offers endpoint calculates the offers in real-time. 28 rate plans are already quite some to calculate offers in real-time. We will have a look at it and see whether we can improve it but, if you are expecting latencies significantly lower than this, you might want to consider the other approach I suggested.
